software/python/src/harp_env_sensor/models/device.py

```diff
@@ -0,0 +1,115 @@
+import logging
```

Comment:
Ooh fancy! This is a nice way to "hydrate" all the registers from the yaml at runtime.

Long term, it would be great to add this in pyharp, but let's keep it here for now.

Reply:
Yeah, I can't actually take credit for this! It's actually Jessy's code from his stage_widget driver. I have started some other improvements to pyharp (such as threaded serial so we can actually catch events), and I'll put this in there too.